tests/samplers_tests/test_nsgaii.py

```diff
@@ -419,24 +419,49 @@ def test_fast_non_dominated_sort_missing_constraint_values() -> None:
     ]
 
 
-def test_crowding_distance_sort() -> None:
-    trials = [
-        _create_frozen_trial(0, [5]),
-        _create_frozen_trial(1, [6]),
-        _create_frozen_trial(2, [9]),
-        _create_frozen_trial(3, [0]),
-    ]
-    optuna.samplers.nsgaii._sampler._crowding_distance_sort(trials)
-    assert [t.number for t in trials] == [2, 3, 0, 1]
+@pytest.mark.parametrize(
+    "values, expected_dist",
+    [
+        ([[5], [6], [9], [0]], [6 / 9, 4 / 9, float("inf"), float("inf")]),
+        ([[5, 0], [6, 0], [9, 0], [0, 0]], [6 / 9, 4 / 9, float("inf"), float("inf")]),
+        (
+            [[5, -1], [6, 0], [9, 1], [0, 2]],
+            [float("inf"), 4 / 9 + 2 / 3, float("inf"), float("inf")],
+        ),
+        ([[5]], [0]),
+        ([[5], [5]], [0, 0]),
+        (
+            [[1], [2], [float("inf")]],
+            [float("inf"), float("nan"), float("inf")],
+        ),  # TODO(knshnb): Decide expected behavior for this case.
+        (
+            [[0], [1], [float("nan")]],
+            [float("inf"), float("nan"), float("inf")],
+        ),  # TODO(knshnb): Decide expected behavior for this case.
+    ],
+)
```

Comment:
Thanks for pointing it out! I agree with the idea that we should only consider the cases that are actually used.
Let me add a comment to the original function on the assumption and modify the test case to meet that.

Reply:
Having said that, I also think it might be still okay to keep the first test case `([[5], [6], [9], [0]], [6 / 9, 4 / 9, float("inf"), float("inf")])` as a simple demonstration of how `_calc_crowding_distance` works. (All values must be the same in one-dimensional cases if they have the same nondomination rank.)
What do you think?

Reply:
I think it's okay to also test the cases that are actually never used if that makes testing more efficient.

Reply:
I don't see how test cases that are not actually used make testing more efficient. Do you have any specific examples?

Reply:
I mean that could make the test cases simpler and easier to read, or fewer tests could cover all branches. The first 1D test case is one example. That case never actually appears but is conceptually simple and easy to understand.

Reply:
I agree with @contramundum53. `_calc_crowding_distance` is decomposed into a sum of distances in each dimension. Although one-dimensional input with different values itself never appears because of the nondomination rank assumption, it appears as a part of multi-dimensional input. So, I think it is efficient to test several one-dimensional cases along with multi-dimensional ones.

Reply:
Thank you for your comments. I agree with you idea. Let me resolve this comment.